spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/channel/PostgresChannelMessageTableSubscription.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.channel;
+
+import org.springframework.integration.jdbc.store.JdbcChannelMessageStore;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.SubscribableChannel;
+import org.springframework.util.Assert;
+
+public final class PostgresChannelMessageTableSubscription {
+
+	private final Object groupId;
+
+	private final String region;
+
+	private final Runnable onPossibleUpdate;
+
+	public PostgresChannelMessageTableSubscription(Object groupId, Runnable onPossibleUpdate) {
+		this(groupId, JdbcChannelMessageStore.DEFAULT_REGION, onPossibleUpdate);
+	}
+
+	public PostgresChannelMessageTableSubscription(Object groupId, String region, Runnable onPossibleUpdate) {
+		this.groupId = groupId;
+		this.region = region;
+		this.onPossibleUpdate = onPossibleUpdate;
+	}
+
+	public Object getGroupId() {
+		return this.groupId;
+	}
+
+	public String getRegion() {
+		return this.region;
+	}
+
+	public void onPossibleUpdate() {
+		this.onPossibleUpdate.run();
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || getClass() != o.getClass()) {
+			return false;
+		}
+		PostgresChannelMessageTableSubscription that = (PostgresChannelMessageTableSubscription) o;
+		if (!this.groupId.equals(that.groupId)) {
+			return false;
+		}
+		if (!this.region.equals(that.region)) {
+			return false;
+		}
+		return this.onPossibleUpdate.equals(that.onPossibleUpdate);
+	}
+
+	@Override
+	public int hashCode() {
+		int result = this.groupId.hashCode();
+		result = 31 * result + this.region.hashCode();
+		result = 31 * result + this.onPossibleUpdate.hashCode();
+		return result;
+	}
+
+	public static SubscribableChannel asSubscribableChannel(
```

Comment:
Why do you still insist on this factory?
This is not Spring-friendly and I doubt anyone is going to use a subscription outside of the channel. 
The `PostgresSubscribableChannel` must be a top-level API and no any extra classes for some `onPossibleUpdate` .
and I doubt in this `onPossibleUpdate()` name yet. Why just not `notify()` ?
What is wrong with extracting contract `Subscription` interface and implement it on the channel?
The `PostgresChannelMessageTableSubscriber` must be a ctor argument for that channel.
Also dispatching must be handled by the `MessageDispatcher` abstraction.

Reply:
My intuition is to expose the most primitive abstraction on the base level, which is receiving a notification whenever a message was added to the group. If you wanted to use this information to read a message from a worker thread or directly, should be up to the user. I made a few changes now to expose a different API that always goes via a `SubscribableChannel`, if you'd prefer not introducing more API. I think this would satisfy the most common use.